src/System.Management.Automation/engine/PSVersionInfo.cs

```diff
@@ -336,7 +336,7 @@ public sealed class SemanticVersion : IComparable, IComparable<SemanticVersion>,
     {
         private const string VersionSansRegEx = @"^(?<major>\d+)(\.(?<minor>\d+))?(\.(?<patch>\d+))?$";
         private const string LabelRegEx = @"^((?<preLabel>[0-9A-Za-z][0-9A-Za-z\-\.]*))?(\+(?<buildLabel>[0-9A-Za-z][0-9A-Za-z\-\.]*))?$";
-        private const string LabelUnitRegEx = @"^[0-9A-Za-z][0-9A-Za-z\-\.]*$";
+        private const string LabelUnitRegEx = @"^[0-9A-Za-z-][0-9A-Za-z\-\.]*$";
```

Comment:
Please add the same update to `LabelRegEx` too. 
And new test for constructor using `LabelRegEx`.

Reply:
Looks like this code was incorrectly using `LabelUnitRegEx` to validate both the prerelease and build labels.  Updated the regex's to match the ones from semver.org (complying with C# syntax).  Fixed the preLabel regex as it's actually using the buildLabel regex previously and added new buildLabel regex.  Fixed the tests so they should be validating correctly now.